core/services/pipeline/common.go

```diff
@@ -299,17 +304,18 @@ func UnmarshalTaskFromMap(taskType TaskType, taskMap interface{}, ID int, dotID
 	}
 
 	decoder, err := mapstructure.NewDecoder(&mapstructure.DecoderConfig{
-		Result: task,
+		Result:           task,
+		WeaklyTypedInput: true,
```

Comment:
I was able to use `WeaklyTypedInput` as a replacement for `string -> int32` and others